File: app/src/main/java/ostmodern/skylark/di/MVPViewsModule.java
package ostmodern.skylark.di;

import dagger.Binds;
import dagger.Module;
import ostmodern.skylark.ui.detail.SetDetailContract;
import ostmodern.skylark.ui.detail.SetDetailFragment;
import ostmodern.skylark.ui.sets.SetListContract;
import ostmodern.skylark.ui.sets.SetListFragment;


@Module
public abstract class MVPViewsModule {

    @Binds
    abstract SetListContract.View provideSetListContractView(SetListFragment setListFragment);

    @Binds
    abstract SetDetailContract.View provideSetDetailContractView(SetDetailFragment setDetailFragment);
}

File: app/src/main/java/ostmodern/skylark/di/MVPViewsModule.kt
package ostmodern.skylark.di

import dagger.Binds
import dagger.Module
import ostmodern.skylark.ui.detail.SetDetailContract
import ostmodern.skylark.ui.detail.SetDetailFragment
import ostmodern.skylark.ui.sets.SetListContract
import ostmodern.skylark.ui.sets.SetListFragment


@Module
abstract class MVPViewsModule {

    @Binds
    internal abstract fun provideSetListContractView(setListFragment: SetListFragment): SetListContract.View

    @Binds
    internal abstract fun provideSetDetailContractView(setDetailFragment: SetDetailFragment): SetDetailContract.View
}
